tensorflow/go/attrs.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tensorflow
+
+// #include <stdlib.h>
+// #include "tensorflow/c/c_api.h"
+//
+// TF_AttrType attrMetadataType(TF_AttrMetadata meta) {
+//   return meta.type;
+// }
+import "C"
+import (
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+// Attr returns info about an attribute on the operation.
+func (op *Operation) Attr(name string) (interface{}, error) {
+	cname := C.CString(name)
+	defer C.free(unsafe.Pointer(cname))
+
+	status := newStatus()
+	meta := C.TF_OperationGetAttrMetadata(op.c, cname, status.c)
+	if err := status.Err(); err != nil {
+		return nil, err
+	}
+	attrType := C.attrMetadataType(meta)
+
+	if meta.is_list == 1 {
```

Comment:
Just to make this easier to read, mind breaking this into two helper functions so this looks something like:

```go
if meta.is_list == 1 {
  return listAttribute(meta)
}
return scalarAttribute(meta)
```

And then these two big switch statements can be in their own helper functions.

Reply:
Done.